Automattic/woocommerce-payments
# Modify 'Contact WooCommerce Support' badge in Payments > Settings page

Fixes #6839 

#### Changes proposed in this Pull Request


This pull request introduces modifications to the 'Contact WooCommerce Support' badge located on the 'Payments accepted on checkout' section of Payments > Settings.

![Screenshot taken on 2023-07-14 at 19 46 31 UTC@2x](https://github.com/Automattic/woocommerce-payments/assets/1553182/02cc56b4-d1ce-4aad-9684-5a8d03a05cc4)

This badge is visible when a payment method cannot be enabled because the merchant needs to provide additional details to Stripe. Since merchants can self-serve this missing details through their connected account dashboard in Stripe, we are modifying the text to inform them that there's missing information, including a link to the public documentation that explains the steps to take. 

<img width="678" alt="Screenshot 2023-07-28 at 18 17 19" src="https://github.com/Automattic/woocommerce-payments/assets/1553182/e514fd45-b6b6-4ae4-8999-7149a4d044da">

In addition to the functional changes mentioned earlier, I'm also taking this opportunity to migrate the involved files to TypeScript (TS). 







#### Testing instructions





* You'll need a Stripe account in a country that requires additional information, for instance Canada.
* Enable BNPLs payment method for the store (feature flag and fees).
* Make sure the UPE payment methods are enabled in the store.
* Navigate to WC Admin > Payments > Settings.
* Check that the new badge appears next to the Afterpay payment method.
* Check that a tooltip appear when you hover the badge.
* Check that the 'Learn more' link in the tooltip takes you to https://woocommerce.com/document/woopayments/payment-methods/buy-now-pay-later/#contact-support.
* Disable UPE payment methods
* Click in 'Enable in your store' button to re-enable them.
* In the Additional payment methods screen, check that the new badge appears and the link takes you to the same documentation page
-------------------


**Post merge**
